### PR TITLE
Tramway Form

### PR DESCRIPTION
**Pull request by kalashnikovisme**
## What's changed basically?

Presenting **useful and convenient** form objects for Rails applications.

*app/forms/user_form.rb
```ruby
class UserForm < Tramway::BaseForm
  properties :email, :password

  def password=(value)
    object.password = value if value.present?
  end
end
```

### Before

*app/controllers/users_controller.rb*
```ruby
class UsersController < ApplicationController
  def create
    @user = User.new

    if @user.save user_params
      render :show
    else
      render :new
    end
  end

  def update
    @user = User.find params[:id]

    if @user.save user_params
      render :show
    else
      render :edit
    end
  end
  
  private

  def user_params
    params[:user].permit!
  end
end
```

### After

*app/controllers/users_controller.rb*
```ruby
class UsersController < ApplicationController
  def create
    @user = tramway_form User.new

    if @user.submit params[:user]
      render :show
    else
      render :new
    end
  end

  def update
    @user = tramway_form User.find params[:id]

    if @user.submit params[:user]
      render :show
    else
      render :edit
    end
  end
end
```

Find more info [here](https://github.com/Purple-Magic/tramway/tree/issue_31#tramway-form)

### CODE REVIEW

File: lib/tramway/base_form.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          @properties || []
+        end
+      end
+    end
+
+    def submit(params)
+      __submit params
+
+      object.save.tap do
+        object.reload
```
Comment (haukot) [inline lib/tramway/base_form.rb:40]: will fail if save didn't happen because model doesn't have id yet

File: lib/tramway/base_form.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      end
+    end
+
+    def method_missing(method_name, *args)
+      if method_name.to_s.end_with?('=') && args.count == 1
+        object.public_send(method_name, args.first)
+      else
+        super
+      end
+    end
```
Comment (haukot) [inline lib/tramway/base_form.rb:58]: `respond_to_missing?` should also be defined

File: lib/tramway/base_form.rb (hunk 1 above)
Comment (kalashnikovisme) [reply lib/tramway/base_form.rb]: Fixed!

Thanks!

File: lib/tramway/base_form.rb (hunk 2 above)
Comment (kalashnikovisme) [reply lib/tramway/base_form.rb:58]: Fixed!

Thanks!

**Pull request merged**